### PR TITLE
#168 Improve testkit for Java usage

### PR DESCRIPTION
**Pull request by anilgursel**

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -29,6 +29,22 @@ class SampleSpec extends CustomTestKit with FlatSpecLike with Matchers {
...
+public class SampleTest extends AbstractCustomTestKit {
+
+    @Test
+    public void testSystemStartTime() {
+        Assert.assertTrue(system().startTime() > 0L);
+    }
+}
+```
+
+The following sections show customizations in ScalaTest; however, all these customizations are supported for TestNG and JUnit as well.  For customizations, provide a `public` constructor in your TestNG/JUnit tests and call `super` with custom parameters.  Check [squbs-testkit/src/test/java/org/squbs/testkit/japi](../squbs-testkit/src/test/java/org/squbs/testkit/japi) for more TestNG and JUnit samples.
```
Comment (akara) [inline docs/testing.md]: Can we just add a quick example on customizations?

**Comment by akara** [comment]
Other than that, LGTM. Thanks!

File: docs/testing.md (hunk 1 above)
Comment (anilgursel) [inline docs/testing.md]: Added a Java sample for customizations.

**Comment by anilgursel** [comment]
I have also addressed some codacy issues in unit tests.

**Pull request merged**